middleware.js

```diff
@@ -1,4 +1,38 @@
 var http = require('http');
 
 exports.createServer = function () {
-};
+  var middlewares = [];
+
+  function next(req, res, index) {
+    if(index == undefined) {
+      index = -1;
+    }
+    return function () {
+      for(var i = index + 1; i < middlewares.length; i++) {
```

Comment:
Yeah, it's my problem. I use FOR loop to match the path and it will find the matched middleware then call it. It works, but the code is difficult to understand. I know how to modify.